src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -96,10 +99,24 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
         String renderedMessage = renderer.render(document);
         
         String result = EmojiParser.parseToUnicode(renderedMessage);	
-		String resultDecimal = EmojiParser.parseToHtmlDecimal(result);	  	
-    %>
-      <li><strong><%= author %>:</strong> <%= resultDecimal %> </li>
-    <%
+		String resultDecimal = EmojiParser.parseToHtmlDecimal(result);
+
+        //uses original str to validate url because the other strings contain extra stuff
+		if (urlValidator.isValid(str)) {
```

Comment:
What does this do with non-image URLs? Something like `http://google.com`?

Maybe consider detecting whether the URL ends with an image extension?

Reply:
also curious about this!

Reply:
Okay guys great suggestion. So basically the code was already taking care of that below in case it was not an image extension. If the image would upload it would display it else it would just show the link as a message. However, I integrated your suggestions so there is a check if the string ends with any image extensions (png, jpg and gif) inside the if statement. I think it is super safe now because it is catching error outside and inside the code block. By the way I also added the GIF extension and it works with gifs as well you can try it out!